pkg/controller/hostpathprovisioner/controller.go

```diff
@@ -44,14 +44,21 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logf.Log.WithName("controller_hostpathprovisioner")
-var watchNamespaceFunc = k8sutil.GetWatchNamespace
+var (
+	log                = logf.Log.WithName("controller_hostpathprovisioner")
+	watchNamespaceFunc = k8sutil.GetWatchNamespace
+)
+
+const (
+	snapshotFeatureGate = "Snapshotting"
+)
 
 func isErrCacheNotStarted(err error) bool {
 	if err == nil {
 		return false
 	}
-	return err.(*cache.ErrCacheNotStarted) != nil
+	_, ok := err.(*cache.ErrCacheNotStarted)
```

Comment:
If we go this way, shouldn't we remove the asterisk?
https://github.com/docker/docker-credential-helpers/blob/master/credentials/error.go#L32-L35

Reply:
No the 'ok' is only true if cache.ErrCacheNotStarted  is not nil, and *cache.ErrCacheNotStarted is true. It has the npe protection build in.

Reply:
Its part of the go type assertion logic.